### PR TITLE
Loosen the pinned fastavro dependency

### PR DESCRIPTION
**Pull request by deleted**
Previously the `fastavro` dependency was pinned to exactly match v1.5.1.

This would make pandavro unable to coexists with other libraries that depend on even slightly newer versions of fastavro, for example [marcosschroh/dataclasses-avroschema](https://github.com/marcosschroh/dataclasses-avroschema)], which requires fastavro v1.5.4.

Loosening this dependency to `~=1.5.1` should allow pandavro to continue working with newer versions of fastavro up until a major version increment.

Tests still appear to pass.

### CODE REVIEW

**Review by ynqa — APPROVED** [review]
LGTM!

**Comment by rabbit72** [comment]
> Loosening this dependency to `~=1.5.1` should allow pandavro to continue working with newer versions of fastavro up until a major version increment.

I believe it means until a minor version of `pandavro` update

```
~=3.1.2: version 3.1.2 or later, but not version 3.2.0 or later.
```
https://peps.python.org/pep-0440/#examples

**Comment by deleted** [comment]
@rabbit72 I stand corrected.
`~=1.5.1` solves my immediate problem. 
I'll leave it up to the project maintainers whether it can be loosened further.

**Pull request merged**
